cmd/controlloop/controlloop.go

```diff
@@ -125,9 +126,19 @@ func handleSignals(stopChannel chan struct{}, signals ...os.Signal) {
 }
 
 func newPodController(stopChannel chan struct{}) (*controlloop.PodController, error) {
-	cfg, err := rest.InClusterConfig()
+	var cfg *rest.Config
+	var err error
+	cfg, err = rest.InClusterConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to implicitly generate the kubeconfig: %w", err)
+		logging.Debugf("failed to generate the kubeconfig from service account: %v", err)
+		kubeConfigFile := os.Getenv("KUBECONFIG")
```

Comment:
both `newPodController` and `ReconcileIPs` are exclusively called from `controlloop.go:main()`.
I know that the operation doesn't cost anything, but it rubs me the wrong way to do `kubeConfigFile := os.Getenv("KUBECONFIG")` in each of them, the correct place imo is in `controlloop.go:main()` and then pass `kubeConfigFile` as an argument to both `newPodController` and `ReconcileIPs` . Perhaps that allows even for more optimizations